packages/lms/src/lib/components/header/Reader.svelte

```diff
@@ -3,65 +3,155 @@
+					isContent = true;
+				}
+			}
 		});
 	});
 
 	$: {
 		// Fetch the audio reactively whenever `content` changes
 		if (content) {
 			(async () => {
-				const response = await fetch(`/api/tts`, {
-					method: 'POST',
-					headers: {
-						'Content-Type': 'application/json'
-					},
-					body: JSON.stringify({ text: content, slug })
-				});
-				const data = await response.json();
-				
+				const data = await fetchAudio(slug);
 				audio.set(data);
+				if (data.exists) {
+					hasAudio = true;
+				} else {
+					hasAudio = false;
+				}
 				audioSrc = data.url;
 			})();
 		}
 	}
+
+	// Regenerate/create the audio file for the current page content
+	async function regenerateAudio() {
+		isloading = true;
+		const filepath = path;
+		const data = await createAudio(content, slug, filepath, true);
+		audio.set(data);
+		audioSrc = data.url;
+		isloading = false;
+		if (data.error) {
+			alert(data.error);
+		} else {
+			alert('Success!, audio file created.');
+		}
+		toggleSettings();
+	}
+
+	// Toggle the settings menu
+	let isOpen = false;
+	function toggleSettings() {
+		isOpen = !isOpen;
+	}
+
+	// For my frustration and amusement
+	let scream: HTMLAudioElement;
+	function playAudio() {
+		scream.play();
+	}
 </script>
 
 <div class="reader">
+	{#if path.startsWith('/content')}
+		<div class="reader-settings">
+			<i class="icon-f reader-settings" role="button" on:click={toggleSettings}>Settings</i>
+			{#if isOpen}
+				<div class="settings-options">
+					{#if hasAudio && !isloading}
+						<button on:click={regenerateAudio}>Regenerate Audio File</button>
+					{:else if isloading}
+						<p>Creating Audio File...</p>
+					{/if}
+					{#if !hasAudio && isContent}
+						<button on:click={regenerateAudio}>Create Audio</button>
+					{/if}
+					<button on:click={playAudio}>Scream For Help!</button>
+					<audio bind:this={scream}>
+						<source src="/public/audio/silly_stuff/female_scream.wav" type="audio/wav" />
+						Your browser does not support the audio element.
+					</audio>
+				</div>
+			{/if}
+		</div>
+	{/if}
 	{#if audioSrc}
 		<audio src={audioSrc} controls class="reader-inner" />
 	{/if}
 </div>
 
 <style>
+	.reader-settings {
+		padding: 0.25rem;
+		color: azure;
+		width: fit-content;
+		position: relative;
+	}
+
+	.settings-options {
+		position: absolute;
+		background-color: var(--primary);
+		color: var(--text-secondary);
+		padding: var(--spacing-m);
+		border-radius: var(--spacing-m);
+		left: 50%;
+		bottom: 100%;
+		border: 1px solid var(--secondary);
+	}
+
+	.settings-options button {
+		display: block;
+		width: 100%;
+		padding: var(--spacing-s);
+		border-radius: var(--spacing-s);
+		background-color: var(--secondary);
+		color: var(--text-primary);
+		margin-bottom: var(--spacing-s);
+		border: none;
+	}
 	.reader {
 		grid-area: reader;
 		width: 100%;
 		margin: 0 auto;
 		text-align: center;
+		display: flex;
 	}
 
 	.reader-inner {
```

Comment:
Can we use Tailwind for some, or all, of this?

Reply:
Yeah, I assume this can be styled upto the skeleton and tailwind stuff after merging, I was just happy to get a branch working on my pc.

Reply:
yeah, I figured the same so I just approved